### ISSUE TITLE
Login not possible for non-admin user

### ISSUE THREAD

**Issue by carstenjaksch** [mollie/Shopware6#704]
A non-admin user can't login. The browser console displays following error message:

```
Failed to load resource: the server responded with a status of 403
/api/_action/system-config?domain=MolliePayments
```

This is the first non-admin user we created with following permissions (ticked all except delete):

- Shopping Experiences
- Categories
- Landingpages
- Media
- Update user profile

Is there any permission a user must have to log in? I tried giving Mollie permissions, but that didn't work.

The Mollie API key is configured per sales channel ("all sales channels" config does not have an API key). I tried to temporarily configure an API key for all sales channels, but that does not solve the issue either.

This line in app.js triggers the error:

```js
S && -1 === ["http", "https", "file"].indexOf(S) ? n(new d("Unsupported protocol " + S + ":",d.ERR_BAD_REQUEST,e)) : b.send(h)
```

Another error in the console while this happens:

```
Uncaught TypeError: Cannot read properties of undefined (reading 'get')
    at Object.SsAR (index.js:5:26)
    at t (custom-pricing.js?1707740867:1:110)
    at Module.snTN (main.js:1:1)
    at t (custom-pricing.js?1707740867:1:110)
    at custom-pricing.js?1707740867:1:953
    at index.js:5:5
```

Code:

```js
if (Shopware.License.get('CUSTOM_PRICES-1673073')) {
    Shopware.Service('privileges').addPrivilegeMappingEntry({
        category: 'permissions',
        parent: null,
        key: 'custom_price',
        roles: {
            viewer: {
                privileges: [
                    'custom_price:read',
                    'product:read',
                    'customer:read',
                    'customer_group:read',
                    'currency:read',
                ],
                dependencies: [],
            },
            editor: {
                privileges: [
                    'custom_price:update',
                ],
                dependencies: [
                    'custom_price.viewer',
                ],
            },
            creator: {
                privileges: [
                    'custom_price:create',
                ],
                dependencies: [
                    'custom_price.viewer',
                    'custom_price.editor',
                    'custom_price.deleter',
                ],
            },
            deleter: {
                privileges: [
                    'custom_price:delete',
                ],
                dependencies: [
                    'custom_price.viewer',
                ],
            },
        },
    });
}
```

**Comment by fjbender** [comment]
Can reproduce on latest Shopware and plugin:

```json
{
    "errors": [
        {
            "status": "403",
            "code": "FRAMEWORK__MISSING_PRIVILEGE_ERROR",
            "title": "Forbidden",
            "detail": "{\"message\":\"Missing privilege\",\"missingPrivileges\":[\"system_config:read\"]}",
            "meta": {
                "parameters": []
            },
            "trace": [
                {
                    "file": "/var/www/html/vendor/shopware/core/Framework/Api/Acl/AclAnnotationValidator.php",
                    "line": 67,
                    "function": "missingPrivileges",
                    "class": "Shopware\\Core\\Framework\\Api\\ApiException",
                    "type": "::"
                },
                {
                    "file": "/var/www/html/vendor/symfony/event-dispatcher/Debug/WrappedListener.php",
                    "line": 116,
                    "function": "validate",
                    "class": "Shopware\\Core\\Framework\\Api\\Acl\\AclAnnotationValidator",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php",
                    "line": 220,
                    "function": "__invoke",
                    "class": "Symfony\\Component\\EventDispatcher\\Debug\\WrappedListener",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/event-dispatcher/EventDispatcher.php",
                    "line": 56,
                    "function": "callListeners",
                    "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/Content/Flow/Dispatching/FlowDispatcher.php",
                    "line": 47,
                    "function": "dispatch",
                    "class": "Symfony\\Component\\EventDispatcher\\EventDispatcher",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/Framework/Webhook/WebhookDispatcher.php",
                    "line": 70,
                    "function": "dispatch",
                    "class": "Shopware\\Core\\Content\\Flow\\Dispatching\\FlowDispatcher",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/Framework/Event/NestedEventDispatcher.php",
                    "line": 31,
                    "function": "dispatch",
                    "class": "Shopware\\Core\\Framework\\Webhook\\WebhookDispatcher",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php",
                    "line": 139,
                    "function": "dispatch",
                    "class": "Shopware\\Core\\Framework\\Event\\NestedEventDispatcher",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpKernel.php",
                    "line": 169,
                    "function": "dispatch",
                    "class": "Symfony\\Component\\EventDispatcher\\Debug\\TraceableEventDispatcher",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpKernel.php",
                    "line": 76,
                    "function": "handleRaw",
                    "class": "Symfony\\Component\\HttpKernel\\HttpKernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/Framework/Adapter/Kernel/HttpKernel.php",
                    "line": 44,
                    "function": "handle",
                    "class": "Symfony\\Component\\HttpKernel\\HttpKernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/Framework/Adapter/Kernel/HttpCacheKernel.php",
                    "line": 50,
                    "function": "handle",
                    "class": "Shopware\\Core\\Framework\\Adapter\\Kernel\\HttpKernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/Kernel.php",
                    "line": 157,
                    "function": "handle",
                    "class": "Shopware\\Core\\Framework\\Adapter\\Kernel\\HttpCacheKernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpCache/SubRequestHandler.php",
                    "line": 86,
                    "function": "handle",
                    "class": "Shopware\\Core\\Kernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpCache/HttpCache.php",
                    "line": 473,
                    "function": "handle",
                    "class": "Symfony\\Component\\HttpKernel\\HttpCache\\SubRequestHandler",
                    "type": "::"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpCache/HttpCache.php",
                    "line": 448,
                    "function": "forward",
                    "class": "Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpCache/HttpCache.php",
                    "line": 346,
                    "function": "fetch",
                    "class": "Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/http-kernel/HttpCache/HttpCache.php",
                    "line": 221,
                    "function": "lookup",
                    "class": "Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/HttpKernel.php",
                    "line": 146,
                    "function": "handle",
                    "class": "Symfony\\Component\\HttpKernel\\HttpCache\\HttpCache",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/shopware/core/HttpKernel.php",
                    "line": 67,
                    "function": "doHandle",
                    "class": "Shopware\\Core\\HttpKernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/public/index.php",
                    "line": 71,
                    "function": "handle",
                    "class": "Shopware\\Core\\HttpKernel",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php",
                    "line": 35,
                    "function": "handle",
                    "class": "Symfony\\Component\\HttpKernel\\HttpKernelInterface@anonymous\u0000/var/www/html/public/index.php:61$0",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/vendor/autoload_runtime.php",
                    "line": 29,
                    "function": "run",
                    "class": "Symfony\\Component\\Runtime\\Runner\\Symfony\\HttpKernelRunner",
                    "type": "->"
                },
                {
                    "file": "/var/www/html/public/index.php",
                    "line": 14,
                    "args": [
                        "/var/www/html/vendor/autoload_runtime.php"
                    ],
                    "function": "require_once"
                }
            ]
        }
    ]
}
```

Not sure if `system_config:read` is something you want to give to a non-admin user?

**Comment by carstenjaksch** [comment]
I tried to disable MolliePayments and the backend works without admin privileges.

Now I can't activate it (backend/console). Error in console:

```
08:16:16 CRITICAL  [console] Error thrown while running command "plugin:activate MolliePayments". Message: "There are 22 error(s) while writing data.\n\n1. [/0/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/1/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/2/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/3/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/4/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/5/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/6/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/7/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/8/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/9/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/10/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/11/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/12/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/13/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/14/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/15/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/16/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/17/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/18/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/19/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/20/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n1. [/21/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein." ["exception" => Shopware\Core\Framework\DataAbstractionLayer\Write\WriteException^ { …},"command" => "plugin:activate MolliePayments","message" => """  There are 22 error(s) while writing data.\n  \n  1. [/0/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/1/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/2/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/3/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/4/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/5/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/6/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/7/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/8/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/9/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/10/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/11/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/12/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/13/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/14/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/15/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/16/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/17/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/18/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/19/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/20/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.\n  1. [/21/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  """]

In WriteContext.php line 43:
                                                                                                   
  There are 22 error(s) while writing data.                                                        
                                                                                                   
  1. [/0/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/1/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/2/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/3/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/4/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/5/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/6/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/7/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/8/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/9/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.   
  1. [/10/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/11/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/12/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/13/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/14/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/15/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/16/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/17/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/18/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/19/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/20/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein.  
  1. [/21/translations/2fbb5fe2e29a4d70aa5854ce7ce3e20b/name] Dieser Wert sollte nicht leer sein. 
```

**Comment by carstenjaksch** [comment]
Any news? We want to go live in a week. As you might guess, this is not possible without payment methods.

**Comment by BlackScorp** [comment]
@carstenjaksch you deleted the system default language. 

if you look in vendor/shopware/core/Defaults.php there is a constant for the system default language.

shopware uses this when you execute code over command line, in command line you dont have a saleschannel and therefore you dont have a default language.

**Comment by carstenjaksch** [comment]
@BlackScorp 

Thanks for the response, but I deleted nothing. The same error happens, when I try to activate Mollie in the backend (see screenshot).

The constant `LANGUAGE_SYSTEM` and the default language in the backend are the same key: 2fbb5fe2e29a4d70aa5854ce7ce3e20b

This extension is the only one with this error message. I assume there are missing translations for `de-DE` that are required by this extension?

<img width="435" alt="Bildschirm­foto 2024-02-21 um 10 40 13" src="https://github.com/mollie/Shopware6/assets/82714642/2b73d0dc-1ffd-4d51-8158-0ddb5bd76515">

**Comment by BlackScorp** [comment]
@carstenjaksch i cannot reproduce this.

when you install the plugin, we update the payment method names or create new payment methods. therefore we use the default shopware repository, nothing special.

https://github.com/mollie/Shopware6/blob/master/src/Service/PaymentMethodService.php#L192

shopware then handles the import. usually you dont need to provide additional language ids for translation.

i assume here that the default language id was changed with another plugin or settings.

if you click on settings > language in admin what is marked as default there?

**Comment by carstenjaksch** [comment]
@BlackScorp The default is `de-DE`. We changed the default language a while back without issues. So how do we fix that? Remove the database entries of Mollie? The only thing we did is connect Shopware and Mollie so that should be no problem.

**Comment by BlackScorp** [comment]
@carstenjaksch set default to en-GB, enable mollie, set default language to de-De again. the constant is fixed inside shopware.

**Comment by carstenjaksch** [comment]
Thank you @BlackScorp, next week is fine.

Will that include a fix for the permission issue this issue was created for initially?

Activating the extension will not help until non-admin users can login again. 🙂

**Comment by BlackScorp** [comment]
@carstenjaksch no, i will not release it until next week, it might take a month or so because of shopware 6.6 so in the mean time, change the default language back to english, enable mollie over console and switch back.

about the non admin user i had no time to look at it

**Comment by BlackScorp** [comment]
@carstenjaksch so i fixed the permission problem and the language problem in the pull request. you can check the code and maybe already apply the patch. iam not sure yet when the next plugin version will be released

**Comment by carstenjaksch** [comment]
@BlackScorp Thank you. I applied the patch and cleared the cache, but activating Mollie results in the same "This value should not be empty" errors.

I did `plugin:refresh` too. Is there anything else I need to do?

<img width="1457" alt="Bildschirm­foto 2024-02-29 um 07 38 16" src="https://github.com/mollie/Shopware6/assets/82714642/f49948f6-c3c7-4d75-9d00-77df7fd869b8">

**Comment by BlackScorp** [comment]
@carstenjaksch ah sorry i used here the wrong key, can you patch again please? i changed the code

**Comment by carstenjaksch** [comment]
@BlackScorp I am sorry to bother you, but unfortunately the error is not fixed. I applied the patch and get the same errors.

I found the steps we used to update the default language back then: [Shopware Forum](https://forum.shopware.com/t/how-to-change-default-language-in-shopware6-from-backend/64220/7)

As I said, this works fine with all installed plugins except for Mollie. Could this information help you fix the "bug"?

**Comment by BlackScorp** [comment]
@carstenjaksch thx ill try to reproduce it

**Comment by carstenjaksch** [comment]
How's it going? Our go live deadline is getting closer.

I tried that again today using the console and with the new version of Mollie, there is an exception trace. Maybe that helps:

<img width="1459" alt="Bildschirm­foto 2024-03-18 um 07 38 19" src="https://github.com/mollie/Shopware6/assets/82714642/bcd9b3a0-351b-4b2f-983a-bb9475f45221">

**Comment by carstenjaksch** [comment]
I have found the issue @BlackScorp. 

File `PaymentMethodService.php`, method `addPaymentMethods`. In the loop, I changed `$paymentMethodData['name']` to `'PaymentMethod' . $key`. Now the plugin can be activated.

The interesting part is, that it is now fixed. I removed my change and the names stay the same. Fortunately, I can change the names back and my changes are saved.

See screenshot for the result:

![Backend](https://github.com/mollie/Shopware6/assets/82714642/9b95bda4-f322-4e12-b1de-dfa8e42a6749)

I think Mollie tried to set the name for some payment methods that haven't been set before and sine we changed the default language ID, it could not find the name or description.

Maybe have a look at this method. I assume we need to do that every time a new payment method is added. So it would be good that Mollie does not use the language ID somehow.
